oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2Configurer.java

```diff
@@ -62,21 +66,41 @@ public void init(HttpSecurity http) throws Exception {
                     http.logout().logoutSuccessHandler(context.getBean(OidcClientInitiatedLogoutSuccessHandler.class));
                 }
 
-            } else {
-                log.debug("OAuth/OIDC Login not configured due to missing issuer, client-id, or client-secret property");
-            }
+                // if issuer is root org, use opaque token validation
+                if (TokenUtil.isRootOrgIssuer(oktaOAuth2Properties.getIssuer())) {
+                    configureResourceServerForOpaqueTokenValidation(http);
+                    return;
+                }
 
-            // resource server configuration
-            if (!context.getBeansOfType(OAuth2ResourceServerProperties.class).isEmpty()) {
-                OAuth2ResourceServerProperties resourceServerProperties = context.getBean(OAuth2ResourceServerProperties.class);
-                if (!isEmpty(resourceServerProperties.getJwt().getIssuerUri())) {
-                    // configure Okta specific auth converter (extracts authorities from `groupsClaim`
-                    configureResourceServer(http, oktaOAuth2Properties);
-                } else {
-                    log.debug("OAuth resource server not configured due to missing issuer property");
+                OAuth2ResourceServerConfigurer oAuth2ResourceServerConfigurer =
+                    http.getConfigurer(OAuth2ResourceServerConfigurer.class);
+
+                if (oAuth2ResourceServerConfigurer != null) {
+
+                    // code below is wrapped in AccessController to address findbugs error 'DP_DO_INSIDE_DO_PRIVILEGED'
+                    //noinspection unchecked
+                    Field jwtConfigurerField = (Field) AccessController.doPrivileged((PrivilegedAction) () -> {
+                        Field result = null;
+                        try {
+                            result = OAuth2ResourceServerConfigurer.class.getDeclaredField("jwtConfigurer");
+                            result.setAccessible(true);
+                        } catch (NoSuchFieldException e) {
+                            log.error("Could not get field 'jwtConfigurer' of {} via reflection",
```

Comment:
nit: if you are continuing is this a warning?

Either way, tell the developer what to do in this case. Maybe mention that this version of Spring Security is not compatible?

Reply:
updated with an apt logger message.